multisense_ros/src/camera.cpp

```diff
@@ -1873,6 +1904,97 @@ void Camera::colorizeCallback(const image::Header& header)
     image_buffers_[header.source] = std::make_shared<BufferWrapper<crl::multisense::image::Header>>(driver_, header);
 }
 
+void Camera::groundSurfaceCallback(const image::Header& header)
+{
+    if (Source_Ground_Surface_Class_Image != header.source)
+    {
+        ROS_WARN("Camera: unexpected image source: 0x%x", header.source);
+        return;
+    }
+
+    const ros::Time t(header.timeSeconds, 1000 * header.timeMicroSeconds);
+
+    switch (header.source)
+    {
+    case Source_Ground_Surface_Class_Image:
+    {
+        const auto ground_surface_subscribers = ground_surface_cam_pub_.getNumSubscribers();
+
+        if (ground_surface_subscribers == 0)
+            return;
+
+        const uint32_t height    = header.height;
+        const uint32_t width     = header.width;
+        const uint32_t imageSize = 3 * height * width;
+
+        ground_surface_image_.data.resize(imageSize);
+
+        ground_surface_image_.header.frame_id = frame_id_rectified_left_;
+        ground_surface_image_.header.stamp    = t;
+        ground_surface_image_.height          = height;
+        ground_surface_image_.width           = width;
+
+        ground_surface_image_.encoding        = sensor_msgs::image_encodings::RGB8;
```

Comment:
Did did not realize were were sending a raw color image over the wire. Does it make sense to send a 8bit mono image where each class has a its unique id. We can then convert that to a RGB image here.

I am approaching this from the angle of saving bandwidth on the wire.

Reply:
Never mind. Should have read the next few lines